timely/src/progress/subgraph.rs

```diff
@@ -94,20 +94,19 @@ where
         self.edge_stash.push((source, target));
     }
 
-    /// Creates a new Subgraph from a channel allocator and "descriptive" indices.
+    /// Creates a new Subgraph from a channel allocator and "descriptive" indices. The Subgraph's
+    /// index is the last element of `path`.
     pub fn new_from(
```

Comment:
I think this is better now! At least, I can't really tell why it made sense to provide the index and the path separately, and there is no `self` involved for which the `path` would make sense. I'm tempted to remove the line in the comment that you added, because "The Subgraph's index" is no longer a concept that anyone needs to know about. I would be tempted to "say more" in the comments, but the previous comments were so vague that it's hard to require it. In fact, .. I think the comments are just wrong? There is no channel allocator here yet.

What do you think about 
```rust
/// Creates a `SubgraphBuilder` from a path of indexes from the dataflow root to the subgraph, terminating with the local index of the new subgraph itself.
```